torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -647,11 +647,30 @@ std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<
 Tensor clamp_backward(const Tensor & grad, const Tensor &self, const optional<Scalar> & min, const optional<Scalar> & max) {
   // clamp: gradients not defined on min and max, so we return the subgradient 1 for these cases.
   if (max && min) {
-    return grad * ((self >= *min) * (self <= *max)).type_as(grad);
+    auto zero = at::scalar_tensor(0., grad.options());
```

Comment:
Why is the backward for the scalar case changing?

Reply:
As @Lezcano pointed out, the old formulation propagates nans from the `grad` even when the gradient is zero.